internal/cmd/commands/database/init.go

```diff
@@ -491,6 +493,64 @@ func (c *InitCommand) ParseFlagsAndConfig(args []string) int {
 	return 0
 }
 
+// initDatabase initializes the schema to the binary's currently suppoorted version.
+// If init previously ran and left any indication that it ran an error is returned.
+// This might be the case for early versions of boundary where the migrations
+// weren't completely executed in a transaction and a version table might have
+// been left despite a rollback.
+// Returns a cleanup function which must be called even if an error is returned and
+// an error code where a non-zero value indicates an error happened.
+func initDatabase(ctx context.Context, ui cli.Ui, dialect, u string) (func(), int) {
```

Comment:
There are only two tiny blocks that differ in this and migrateDatabase. I'm not sure why you pulled it out into two functions that are essentially the same, but could it be put back and simply use a bool to control if it's init vs. migrate behavior?

Reply:
Fixed this.